### ISSUE TITLE
Dynamically resolved helpers don't work from partials

### ISSUE THREAD

**Issue by markkimsal** [zordius/lightncandy#251]
The PHP Code:
-------------

```php
require('./vendor/autoload.php');
use LightnCandy\LightnCandy;

// Non-working template:
$template = <<<VAREND
{{>baz}}
VAREND;

//working template
/*
$template = <<<VAREND
{{someUndefHelper abc='foo'}}
VAREND;
*/

// Partials:
$partials = array(
    'baz' => 'Yes, partial  {{someUndefHelper abc=\'foo\'}}',
);

function someUndefHelper($options) { 
return "bar";
}
    
$phpStr = LightnCandy::compile($template, array(
  // Used compile flags
  'flags' => LightnCandy::FLAG_ERROR_EXCEPTION | LightnCandy::FLAG_HANDLEBARS |
            LightnCandy::FLAG_RUNTIMEPARTIAL |
            LightnCandy::FLAG_EXTHELPER |
            LightnCandy::FLAG_HANDLEBARSJS,
            
  'partials' => $partials,
  'helperresolver'=> function(&$cx, $name) {
    if (array_key_exists($name,  $cx['helpers'])) {
        return;
    }   
    //$cx['helpers'][$name] = $name;
    return $name;
  } 
)); 

echo "Generated PHP Code:\n$phpStr\n";

// Input Data:
$data = array(
  'qoo' => 'moo'
);

// Save the compiled PHP code into a php file
file_put_contents('render.php', '<?php ' . $phpStr . '?>');

// Get the render function from the php file
$renderer = include('render.php');

echo "Result:\n" . $renderer($data);
```

The Issue:
----------

Helpers resolved with a helperresolver that appear only in partials do not end up in the context's helpers array, but they do end up in the usageCount -> helpers array.

Calling {{someUndefHelper}} from the main template works, but calling it only from the partial doesn't create a reference in the helpers array in the compiled code.

**Comment by jayS-de** [comment]
Your helper resolver doesn't make much sense in combination with FLAG_EXTHELPER

When I remove your helper resolver the rest of the code above works

**Comment by jayS-de** [comment]
Found the probably a solution to this issue:

In Context.php the merge function is missing a merge for the helpers

```
class Context extends Flags {
  ...
    public static function merge(&$context, $tmp) {
        $context['error'] = $tmp['error'];
        $context['helpers'] = $tmp['helpers']; // <-- this line is missing
        $context['partials'] = $tmp['partials'];
        $context['partialCode'] = $tmp['partialCode'];
        $context['partialStack'] = $tmp['partialStack'];
        $context['usedCount'] = $tmp['usedCount'];
        $context['usedFeature'] = $tmp['usedFeature'];
        $context['usedPartial'] = $tmp['usedPartial'];
    }
}
```

**Comment by zordius** [comment]
Thank you for the report and the solution! Will work on this later, thanks!

**Comment by zordius** [comment]
This should be fixed in current master branch, please try again, thanks.

**Comment by jayS-de** [comment]
Works 👍

**Comment by zordius** [comment]
released in 1.0.1
